Common/ModdedPlayer.cs

```diff
@@ -16,326 +16,263 @@
-                Player.QuickSpawnItem(null, ItemID.QueenSlimeCrystal);
-            }
-            if (NPC.AnyNPCs(NPCID.Spazmatism) || NPC.AnyNPCs(NPCID.Retinazer))
-            {
-                Player.QuickSpawnItem(null, ItemID.MechanicalEye);
-            }
-            if (NPC.AnyNPCs(NPCID.TheDestroyer))
-            {
-                Player.QuickSpawnItem(null, ItemID.MechanicalWorm);
-            }
-            if (NPC.AnyNPCs(NPCID.SkeletronPrime))
-            {
-                Player.QuickSpawnItem(null, ItemID.MechanicalSkull);
-            }
-            if (NPC.AnyNPCs(NPCID.Plantera))
-            {
-                Player.QuickSpawnItem(null, ModContent.ItemType<PlanteraSpawn>());
-            }
-            if (NPC.AnyNPCs(NPCID.Golem))
-            {
-                Player.QuickSpawnItem(null, ItemID.LihzahrdPowerCell);
-            }
-            if (NPC.AnyNPCs(NPCID.DukeFishron))
-            {
-                Player.QuickSpawnItem(null, ItemID.TruffleWorm);
-            }
-            if (NPC.AnyNPCs(NPCID.HallowBoss))
-            {
-                Player.QuickSpawnItem(null, ItemID.EmpressButterfly);
-            }
-            if (NPC.AnyNPCs(NPCID.CultistBoss))
-            {
-                Player.QuickSpawnItem(null, ModContent.ItemType<LunaticCultistSpawner>());
-            }
-            if (NPC.AnyNPCs(NPCID.MoonLordCore))
-            {
-                Player.QuickSpawnItem(null, ItemID.CelestialSigil);
-            }
-        }
-        public override void SyncPlayer(int toWho, int fromWho, bool newPlayer)
-        {
-            ModPacket packet = Mod.GetPacket();
-            packet.Write((byte)BossRush.MessageType.GodUltimateChallenge);
-            packet.Write((byte)Player.whoAmI);
-            packet.Write(gitGud);
-            packet.Send(toWho, fromWho);
-        }
-        public override void Initialize()
-        {
-            gitGud = 0;
-        }
-        public override void SaveData(TagCompound tag)
-        {
-            tag["gitgud"] = gitGud;
-        }
-        public override void LoadData(TagCompound tag)
-        {
-            gitGud = (int)tag["gitgud"];
-        }
-        public void ReceivePlayerSync(BinaryReader reader)
-        {
-            gitGud = reader.ReadInt32();
-        }
+		public override void Kill(double damage, int hitDirection, bool pvp, PlayerDeathReason damageSource) {
+			SpawnItem();
+			if (ModContent.GetInstance<BossRushModConfig>().Nightmare) {
+				DeleteCardItem();
+			}
+		}
+		private void DeleteCardItem() {
+			for (int i = 0; i < Main.item.Length; i++) {
+				Item item = Main.item[i];
+				if (item is null || item.ModItem is null) {
+					continue;
+				}
+				if (item.ModItem is CardPacketBase) {
+					item.active = false;
+				}
+			}
+			foreach (var item in Player.inventory) {
+				if (item is null || item.ModItem is null) {
+					continue;
+				}
+				if (item.ModItem is CardPacketBase) {
+					item.stack = 0;
+					item.active = false;
```

Comment:
Do not use or set `item.active` outside world context.
You should prefer using `TurnToAir` method instead.